NPE in TypeUtils.hasAnnotationOfType instead of compilation error

Lately we moved our BaseObject class that is extended by our actual `@Entity` classes to another package. It wasn't possible to do it fully via safe IDE refactorings and we ended up with couple of imports in entity classes pointing to old BaseObject (unknowingly, of course). Trouble was, that error message was more than cryptic:
```
[INFO] --- apt-maven-plugin:1.1.1:process (default) @ querydsl-npe ---
[ERROR] execute error
java.lang.RuntimeException: java.lang.NullPointerException
	at com.sun.tools.javac.main.Main.compile(Main.java:553)
	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
	at com.mysema.maven.apt.AbstractProcessorMojo.execute(AbstractProcessorMojo.java:314)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:101)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:209)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:84)
	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:59)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.singleThreadedBuild(LifecycleStarter.java:183)
	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:161)
	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:320)
	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:156)
	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:537)
	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:196)
	at org.apache.maven.cli.MavenCli.main(MavenCli.java:141)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:290)
	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:230)
	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:409)
	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:352)
	at org.codehaus.classworlds.Launcher.main(Launcher.java:47)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
Caused by: java.lang.NullPointerException
	at com.mysema.query.apt.TypeUtils.hasAnnotationOfType(TypeUtils.java:48)
	at com.mysema.query.apt.TypeUtils.hasAnnotationOfType(TypeUtils.java:40)
	at com.mysema.query.apt.AbstractQuerydslProcessor.addExternalParents(AbstractQuerydslProcessor.java:186)
	at com.mysema.query.apt.AbstractQuerydslProcessor.processAnnotations(AbstractQuerydslProcessor.java:158)
	at com.mysema.query.apt.AbstractQuerydslProcessor.process(AbstractQuerydslProcessor.java:97)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
	at com.sun.tools.javac.main.Main.compile(Main.java:523)
	... 30 more
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
```
I checked the setup, no problem. So I debugged the build with breakpoint on the critical line and the trouble is that variable element is indeed null. Two stackframes higher (addExternalParents():186, AbstractQuerydslProcessor (com.mysema.query.apt)) I was able to find the offending class (our entity) and found the problematic import.

While it is our fault, is it possible to enhance the diagnostic message?

For your convenience I provide project for reproduction: https://github.com/virgo47/litterbin/tree/master/issues/querydsl-npe

Easiest way to do reproduce:
```
$ svn co https://github.com/virgo47/litterbin/trunk/issues/querydsl-npe
$ cd querydsl-npe/
$ mvn package
```

--- a/querydsl-apt/src/main/java/com/querydsl/apt/AbstractQuerydslProcessor.java
+++ b/querydsl-apt/src/main/java/com/querydsl/apt/AbstractQuerydslProcessor.java
@@ -173,11 +173,11 @@ public abstract class AbstractQuerydslProcessor extends AbstractProcessor {
             Type superType = superTypes.pop();
             if (!context.allTypes.containsKey(superType.getFullName())) {
                 TypeElement typeElement = processingEnv.getElementUtils().getTypeElement(superType.getFullName());
-                if (conf.isStrictMode() && !TypeUtils.hasAnnotationOfType(typeElement, conf.getEntityAnnotations())) {
-                    continue;
-                }
                 if (typeElement == null) {
                     throw new IllegalStateException("Found no type for " + superType.getFullName());
+                }
+                if (conf.isStrictMode() && !TypeUtils.hasAnnotationOfType(typeElement, conf.getEntityAnnotations())) {
+                    continue;
                 }
                 EntityType superEntityType = elementHandler.handleEntityType(typeElement);
                 if (superEntityType.getSuperType() != null) {

